### ISSUE TITLE
Map settings reusability improvement suggestion

### ISSUE THREAD

**Issue by TheLeonsver1** [StarArawn/bevy_ecs_tilemap#8]
Currently `Map` isn't `Clone` (and I don't think it should be, or at least not for this issue's purposes) 
and `Map::new()` currently takes:
```
size,
chunk_size,
tile_size,
texture_size,
(layer_id isn't relevant to this issue)
```

My suggestion would be to create a `MapSettings` struct that is `Clone`/`Copy`:
```rust
///The settings for tilemap layers
#[derive(Debug, Clone, Copy)]
pub struct MapSettings {
    pub map_size: MapVec2,
    pub chunk_size: MapVec2,
    pub tile_size: Vec2,
    pub texture_size: Vec2,
}
```
To enable making similar maps with the same settings more convenient(for example if you need layers with the same settings). The current impl kind of forces the user to create functions to reduce boilerplate which I think should be avoided)

This also enables storing these structs as/in resources for reusability
`Map::new()` would then take `MapSettings` instead of the 4 different values

*I don't mean you need to actually store this inside of `Map` instead of the current configuration. 
you can and maybe should keep `Map` as is and only change `Map::new()` to make it more convenient for users. I'm not sure.. it might be nice to copy the settings of a `Map` if it's possible, idk if there'd be uses for that, maybe for `bevy_tiled` maps..?

**Comment by StarArawn** [comment]
Thanks, this is a great feature suggestion! 👍 I don't think I want `Map` to be clone either, but I do agree having an easier way of cloning map settings would be great! I'm currently working through some of the backend chunk stuff to make things more performant, either you can make a PR for this or I can add this feature sometime in the near future.

**Comment by TheLeonsver1** [comment]
closed since it was done in 3ac42b4
